### ISSUE TITLE
rustdoc doesn't produce "Trait Implementations" for type aliases

### ISSUE THREAD

**Issue by jbapple** [rust-lang/rust#19381]
```
pub trait Foo {}
pub struct Bar;
pub type Baz = Bar;
impl Foo for Baz {}
pub trait Qux {}
impl Qux for Bar {}
```

For this, rustdoc produces a "Trait Implementations" section for `Bar`, but not for `Baz`. Furthermore, `Baz`'s "Trait Implementations" section lists only `Qux`, not `Foo`. The "Implementors" section on the `Foo` page does list that `Baz` is an implementor.

**Comment by steveklabnik** [comment]
Triage: no change.

**Comment by QuietMisdreavus** [comment]
Fixed by #42027.
